dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -27,17 +27,60 @@
 # *****************************************************************************
 
 
+import dpctl
 import dpctl.tensor._tensor_impl as ti
+from numpy import prod
 
 import dpnp
 import dpnp.backend.extensions.lapack._lapack_impl as li
 
-__all__ = ["dpnp_eigh"]
+__all__ = ["dpnp_eigh", "dpnp_solve"]
 
 _jobz = {"N": 0, "V": 1}
 _upper_lower = {"U": 0, "L": 1}
 
 
+def _common_type(*arrays):
+    """
+    _common_type(*arrays)
+
+    Common type for linalg
+
+    This function determines the common data type for linalg operations.
+    It's designed to be similar in logic to `numpy.linalg.linalg._commonType`.
+
+    Key differences from `numpy.common_type`:
+    - It accepts ``bool_`` arrays.
+    - It doesn't consider ``float16`` arrays since they are not supported.
```

Comment:
`float16` is acceptable by dpctl, since we will cast input array anyway, is there any reason to prohibit `float16` arrays?

Reply:
That line should be deleted
Thank you for noticing